packages/protocol-sdk/src/mint-client.ts

```diff
@@ -0,0 +1,212 @@
+      },
+      { token_id: tokenId?.toString() },
+    );
+
+    if (!mintable.feed_item.is_active) {
+      throw new MintInactiveError("Minting token is inactive");
+    }
+
+    if (!mintable.feed_item.mint_context) {
+      throw new MintError("No minting context data from zora API");
+    }
+
+    if (
+      !["zora_create", "zora_create_1155"].includes(
+        mintable.feed_item.mint_context?.mint_context_type!,
+      )
+    ) {
+      throw new MintError(
+        `Mintable type ${mintable.feed_item.mint_context.mint_context_type} is currently unsupported.`,
+      );
+    }
+
+    if (
+      mintable.feed_item.mint_context.mint_context_type === "zora_create_1155"
+    ) {
+      return {
+        send: await this.mintZora1155({
+          publicClient: this.getPublicClient(publicClient),
+          mintArguments,
+          sender,
+          mintable,
+        }),
+        mintable,
+      };
+    }
+    if (mintable.feed_item.mint_context.mint_context_type === "zora_create") {
+      return {
+        send: await this.mintZora721({
+          publicClient: this.getPublicClient(publicClient),
+          mintArguments,
+          sender,
+          mintable,
+        }),
+        mintable,
+      };
+    }
+
+    throw new Error("Mintable type not found or recognized.");
+  }
+
+  private async mintZora1155({
+    mintable,
+    sender,
+    publicClient,
+    mintArguments,
+  }: MintParameters) {
+    const mintQuantity = BigInt(mintArguments.quantityToMint);
+
+    const address = mintable.feed_item.collection.address as Address;
+
+    const mintFee = await publicClient.readContract({
+      abi: zoraCreator1155ImplABI,
+      functionName: "mintFee",
+      address,
+    });
+
+    const tokenFixedPriceMinter = await this.apiClient.getSalesConfigFixedPrice(
+      {
+        contractAddress: mintable.feed_item.contract_address,
+        tokenId: mintable.feed_item.token_id!,
+        chainId: this.network.chainId,
+      },
+    );
+
+    return async (walletClient: WalletClient) => {
+      const { request } = await publicClient.simulateContract({
+        abi: zoraCreator1155ImplABI,
+        functionName: "mintWithRewards",
+        account: sender,
+        value:
+          (mintFee + BigInt(mintable.feed_item.cost.native_price.raw)) *
+          mintQuantity,
+        address,
+        /* args: minter, tokenId, quantity, minterArguments, mintReferral */
+        args: [
+          (tokenFixedPriceMinter ||
+            zoraCreatorFixedPriceSaleStrategyAddress[999]) as Address,
+          BigInt(mintable.feed_item.token_id!),
+          mintQuantity,
+          encodeAbiParameters(parseAbiParameters("address, string"), [
+            mintArguments.mintToAddress,
+            mintArguments.mintComment || "",
+          ]),
+          mintArguments.mintReferral || zeroAddress,
+        ],
+      });
+      return await walletClient.writeContract(request);
+    };
+  }
+
+  private async mintZora721({
```

Comment:
this PR is a bit big to review, could you break out the mint functionality into its own PR (separate from creation of tokens)